### PR TITLE
Tell an NPC to finish your construction tasks

### PR DESCRIPTION
**Pull request by ghost**
#### Summary
```SUMMARY: Features "Tell an NPC to finish your construction tasks"```

#### Purpose of change

Adds an NPC talk function to allies, that tells them to go and work on your unfinished constructions ( if they have the skill )

#### Describe the solution
Adds a multiple_construction activity, that makes them go from construction to construction , finishing them off.
You can also work on the same construction at the same time as the NPC, working together! and then itll be finished quicker.

#### Describe alternatives you've considered
This will be expanded in the future to be a camp mission as well as the NPC starting the construction themselves based on designated blueprint areas.

#### Additional context
Currently, there is no check for tools still being available during the activity, am waiting for Ifreunds implementation of that for unfinished crafts before I can shamelessly steal that work.

Instead, we can pretend that the unfinished construction is left in a state that an NPC can just walk up and work on it, without the need for appropiate tools , or maybe tht the player throwsthem the tools as needed.

Currently, NPCS can not learn from watching other NPCs doing the activity, and do not assist them.
Thats probably fairly simple to implement, but its not essential right now.

Overall, this PR was surprisingly easy to do, and thats making me suspicious that I've missed something, I've tested a number of different permutations and it just seems to fall into place... so we shall see.

### CODE REVIEW

**Comment by I-am-Erk** [comment]
How far a step is it from this to having a static npc in a place like the refugee center run around automatically fixing up broken windows and things?

File: data/json/player_activities.json [hunk 1]
```diff
@@ -16,6 +16,15 @@
     "based_on": "speed",
     "refuel_fires": true
   },
+  {
+  "id": "ACT_MULTIPLE_CONSTRUCTION",
```
Comment (anothersimulacrum) [inline data/json/player_activities.json]: This needs to be linted

**Comment by davidpwbrown** [comment]
> How far a step is it from this to having a static npc in a place like the refugee center run around automatically fixing up broken windows and things?

Hmmm, thatd be a bit different, if there is a construction task to repair things, that have a broken state....
For windows itd be - if is broken, then clean broken window, then build new one.
And id need to implement them *starting* their own construction activities instead of just resuming ones the player started.
So 50% there id say but some tricky stuff to do before thats possible.

**Comment by I-am-Erk** [comment]
Fair enough.

For repair actions like that I think you'd mostly want them to go through the current "nati

**Comment by I-am-Erk** [comment]
Gah hit the wrong button

**Comment by davidpwbrown** [comment]
> Gah hit the wrong button

HAH how dare you!!! :P

**Comment by I-am-Erk** [comment]
Who will stop me, Hmm? *You*?

**Comment by sashakiller33** [comment]
Maybe it would be better to make it more like an area for such task? Where is an area there they will repair broken buildings, and there are already some area works framework.

**Comment by I-am-Erk** [comment]
We'll probably need some kind of reference mapgen file that the NPC follows actually.

**Pull request merged**